Baby Monitor/Source Files/Services/Connection/SocketCommunicationManager.swift

```diff
@@ -1,37 +1,46 @@
 import Foundation
 import RxSwift
 
-protocol SocketCommunicationManager: class {
+protocol SocketCommunicationManager: class, WebSocketConnectionStatusProvider {
     var communicationResetObservable: Observable<Void> { get }
     var communicationTerminationObservable: Observable<Void> { get }
     func reset()
     func terminate()
 }
 
 class DefaultSocketCommunicationManager: SocketCommunicationManager {
-    private(set) var communicationResetObservable: Observable<Void>
-    private(set) var communicationTerminationObservable: Observable<Void>
     private var communicationResetPublisher = PublishSubject<Void>()
+    var communicationResetObservable: Observable<Void> {
+        return communicationResetPublisher.asObservable()
+    }
     private var communicationTerminationPublisher = PublishSubject<Void>()
+    var communicationTerminationObservable: Observable<Void> {
+        return communicationTerminationPublisher.asObservable()
+    }
+    private var socketConnectionStatusPublisher = PublishSubject<WebSocketConnectionStatus>()
+    var connectionStatusObservable: Observable<WebSocketConnectionStatus> {
```

Comment:
`Private` vars should be below internal ones.

Reply:
It's not a private getter - it's exposed eg. to DashboardViewModel to bind "connected" control with the state of webCocket communication

Reply:
I mean that in this files private and internal variables are mixed 😄

Reply:
I thought that is a convention in the project - mix private publisher with its observable being a dynamic getter... Fixed it anyway ;)